src/graph/unit_graph.cc

```diff
@@ -897,13 +1039,26 @@ HeteroGraphPtr UnitGraph::CopyTo(HeteroGraphPtr g, const DLContext& ctx) {
   //   be fixed later.
   auto bg = std::dynamic_pointer_cast<UnitGraph>(g);
   CHECK_NOTNULL(bg);
+
   CSRPtr new_incsr = CSRPtr(new CSR(bg->GetInCSR()->CopyTo(ctx)));
   CSRPtr new_outcsr = CSRPtr(new CSR(bg->GetOutCSR()->CopyTo(ctx)));
-  return HeteroGraphPtr(new UnitGraph(g->meta_graph(), new_incsr, new_outcsr, nullptr));
+  return HeteroGraphPtr(
+      new UnitGraph(g->meta_graph(), new_incsr, new_outcsr, nullptr, SparseFormat::ANY));
```

Comment:
Same here.

Reply:
Besides, we don't have a GPU implementation for COOToCSR, so message passing on GPUs will fail.